LinqChallengesFivePack/LinqChallengesFivePack/LinqChallenges.cs

```diff
@@ -44,17 +52,24 @@ public double CalculateAlbumDurationSeconds()
         //Output should be: "0,0" "0,1" "0,2" "1,0" "1,1" "1,2" "2,0" "2,1" "2,2"
         public List<string> CalculateMatrixPoints3x3()
         {
-            
-            return new List<string>();
+            int n = 3; // 3 x 3;
+            var answer = Enumerable.Range(0, n)
+                .SelectMany(x => Enumerable.Range(0, n).Select(y => $"{x},{y}"))
+                .ToList();
+            return answer;
         }
 
         //Given the input, return a collection of integers with the ranges filled in
         //Solution should be: 2, 5, 7, 8, 9, 10, 11, 17, 18
         public List<int> GetRangesFromString()
-        {  
+        {
             var input = "2,5,7-10,11,17-18";
-
-            return new List<int>();
+            var answer = input.Split(',')
+                .SelectMany(i => i.Split('-').Length == 1 
+                ? new List<int> { int.Parse(i) }
+                : Enumerable.Range(int.Parse(i.Split('-')[0]), 1 + int.Parse(i.Split('-')[1]) - int.Parse(i.Split('-')[0])))
```

Comment:
GENIUS. Funny thing is I actually used `Enumerable.Range` in one of my other answers but didn't think about using it here. Nice.